### PR TITLE
respond bad gateway correctly fixes #3159

### LINKED ISSUE #3159: ERROR yamlconf: yamlconf.go:260 Custom polling failed: Unable to open gzip reader: gzip: invalid header

### PR DESCRIPTION
**Pull request by fffw**
@myleshorton It would be good to also including this fix in new release.

### CODE REVIEW

File: src/github.com/getlantern/flashlight/client/handler.go [hunk 1]
```diff
@@ -170,18 +170,11 @@ func respondOK(writer io.Writer, req *http.Request) error {
...
-		StatusCode: http.StatusBadGateway,
-		ProtoMajor: 1,
-		ProtoMinor: 1,
-	}
-	err := resp.Write(w)
-	if err == nil {
-		if _, err = w.Write([]byte(msg)); err != nil {
-			log.Debugf("Error writing error to io.Writer: %s", err)
-		}
+	resp.WriteHeader(http.StatusBadGateway)
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/client/handler.go]: It's Intended to return HTTP 502, but it actually return 200 OK with the response as content. Fixed it.

File: src/github.com/getlantern/flashlight/client/handler_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package client
+
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/client/handler_test.go]: Wanted to add test but failed. Leave it to implement later.

File: src/github.com/getlantern/flashlight/flashlight_test.go [hunk 3]
```diff
@@ -196,7 +196,7 @@ func testRequest(testCase string, t *testing.T, requests chan *http.Request, htt
 	} else if requestSuccessful {
 		defer func() {
 			if err := resp.Body.Close(); err != nil {
-				t.Fatalf("Error closing response body", err)
+				t.Fatalf("Error closing response body: %s", err)
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/flashlight_test.go]: Oh it should have fixed in some other PR, may need to resolve conflict when merging.

**Comment by myleshorton** [comment]
LGTM!

**Pull request merged**